.github/workflows/ci.yml

```diff
@@ -59,49 +96,263 @@ jobs:
       if: steps.cachedlk.outputs.cache-hit != 'true'
       run: |
         cd $GITHUB_WORKSPACE/lk
-        cmake -Bbuild_linux -DCMAKE_BUILD_TYPE=Debug
-        cmake --build build_linux -- -j
-    - name: Set LKDIR
+        cmake -Bbuild_linux -DCMAKE_BUILD_TYPE=Release
+        cmake --build build_linux -j4
+    - name: Set env vars
       run: |
         echo "LKDIR=$GITHUB_WORKSPACE/lk" >>$GITHUB_ENV
-    - name: Set LKD_LIB 
-      run: |
-        echo "LKD_LIB=$GITHUB_WORKSPACE/lk/build_linux" >>$GITHUB_ENV
-    - name: Set RAPIDJSONDIR
-      run: |
+        echo "LK_LIB=$GITHUB_WORKSPACE/lk/build_linux" >>$GITHUB_ENV
         echo "RAPIDJSONDIR=$GITHUB_WORKSPACE/ssc" >>$GITHUB_ENV
     - name: Get git ref of sibling dependency SSC
       run: |
-        ref=$(git ls-remote --exit-code git://github.com/NREL/ssc.git refs/heads/patch | awk '{print $1}')
+        ref=$(git ls-remote --exit-code https://github.com/NREL/ssc.git refs/heads/develop | awk '{print $1}')
```

Comment:
This looks like there will be a branch mismatch between patch and develop. Should this be patch?

Reply:
For events triggered by `pull_request`, then you have the pull request merge branch, but for the standalone `push`, how should the default branch be determined as develop or patch?

Reply:
Currently in SAM's patch branch, the base branch is also develop: https://github.com/NREL/SAM/blob/1ed59e9b181cf1f23d7ac1f696b5cb91805c7936/.github/workflows/ci.yml#L70

Reply:
That's not intentional: https://github.com/NREL/SAM/pull/1664 I'm not sure how that got reverted but it should be patch.

Reply:
> For events triggered by `pull_request`, then you have the pull request merge branch, but for the standalone `push`, how should the default branch be determined as develop or patch?

Branches from patch should have patch as the default in the CI script, whereas branches from develop should have develop.

Reply:
> > For events triggered by `pull_request`, then you have the pull request merge branch, but for the standalone `push`, how should the default branch be determined as develop or patch?
> 
> Branches from patch should have patch as the default in the CI script, whereas branches from develop should have develop.

Yes, that is clear. My question was mechanistic. I may just have two versions of ci.yml on each branch.

Reply:
Most years, a commit when we switch from patch to develop (and vice versa) has been sufficient, but as we see right now it's error prone. How would two ci.yml instances work?

Reply:
This might be a bad idea but I was thinking we'd have a different version of ci.yml on each branch. I think then the first time we merge patch into develop, we have to manually unselect the changes in ci.yml. But then after that, that difference won't be a tracked diff, and shouldn't show up. I'm not 100% sure it'll work but I could try.

There kind of is a way to get the "closest parent branch" but it uses a lot of bash magic (that I don't understand) and might be unreliable.